### DISCUSSION TITLE
Error when trying to test component using useQuery from geschichte

### DISCUSSION THREAD

**Discussion by Morritz** [BowlingX/geschichte#38] [Q&A] [answered]
```
TypeError: useStore is not a function
 ❯ Module.useQuery node_modules/geschichte/build/mjs/lib/store.js:59:97
```

Steps to reproduce.
1. Start vite project with react and typescript
2. Install vitest and react testing library
3. Try to render component with useQuery in test file.

**Comment by BowlingX** [comment, marked as answer]
Thank you for your Question.

You have to provide the `<GeschichteContextProvider />` in your testing render tree.
Any hooks (e.g. useQuery etc) cannot be used without the Provider.

If you use `history` you can have an example how I did it for the library tests here: 
https://github.com/BowlingX/geschichte/blob/master/src/__tests__/index.tsx#L101

According to your Feedback this error is now more clear in version `10.0.2`.
